src/sonic-yang-mgmt/_sonic_yang_ext.py

```diff
@@ -0,0 +1,616 @@
+
+        assert topLevelContainer['@name'] == moduleName
+
+        container = topLevelContainer['container']
+        # container is a list
+        if isinstance(container, list):
+            for c in container:
+                self.confDbYangMap[c['@name']] = {
+                    "module" : moduleName,
+                    "topLevelContainer": topLevelContainer['@name'],
+                    "container": c
+                    }
+        # container is a dict
+        else:
+            self.confDbYangMap[container['@name']] = {
+                "module" : moduleName,
+                "topLevelContainer": topLevelContainer['@name'],
+                "container": container
+                }
+    return
+
+"""
+Get module, topLevelContainer and json container for a config DB table
+"""
+def get_module_top_container(self, table):
+    cmap = self.confDbYangMap
+    m = cmap[table]['module']
+    t = cmap[table]['topLevelContainer']
+    c = cmap[table]['container']
+    return m, t, c
+
+"""
+Crop config as per yang models
+"""
+def cropConfigDB(self, croppedFile=None):
+
+    for table in self.jIn.keys():
+        if table not in self.confDbYangMap:
+            del self.jIn[table]
+
+    if croppedFile:
+        with open(croppedFile, 'w') as f:
+            dump(self.jIn, f, indent=4)
+
+    return
+
+"""
+Extract keys from table entry in Config DB and return in a dict
+"""
+def extractKey(self, tableKey, regex):
+
+    # get the keys from regex of key extractor
+    keyList = re.findall(r'<(.*?)>', regex)
+    # create a regex to get values from tableKey
+    # and change separator to text in regexV
+    regexV = re.sub('<.*?>', '(.*?)', regex)
+    regexV = re.sub('\|', '\\|', regexV)
+    # get the value groups
+    value = re.match(r'^'+regexV+'$', tableKey)
+    # create the keyDict
+    i = 1
+    keyDict = dict()
+    for k in keyList:
+        if value.group(i):
+            keyDict[k] = value.group(i)
+        else:
+            raise Exception("Value not found for {} in {}".format(k, tableKey))
+        i = i + 1
+
+    return keyDict
+
+"""
+Fill the dict based on leaf as a list or dict @model yang model object
+"""
+def fillLeafDict(self, leafs, leafDict, isleafList=False):
+
+    if leafs == None:
+        return
+
+    # fill default values
+    def fillSteps(leaf):
+        leaf['__isleafList'] = isleafList
+        leafDict[leaf['@name']] = leaf
+        return
+
+    if isinstance(leafs, list):
+        for leaf in leafs:
+            #print("{}:{}".format(leaf['@name'], leaf))
+            fillSteps(leaf)
+    else:
+        #print("{}:{}".format(leaf['@name'], leaf))
+        fillSteps(leafs)
+
+    return
+
+"""
+create a dict to map each key under primary key with a dict yang model.
+This is done to improve performance of mapping from values of TABLEs in
+config DB to leaf in YANG LIST.
+"""
+def createLeafDict(self, model):
+
+    leafDict = dict()
+    #Iterate over leaf, choices and leaf-list.
```

Comment:
Only need to handle these 3 types?

Reply:
As per RFC 7950, yes. But for sure, new cases may come later.